lib/player.js

```diff
@@ -7094,6 +7094,7 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       oldTrack = shaka.util.StreamUtils.variantToTrack(currentVariant);
     }
     const newTrack = shaka.util.StreamUtils.variantToTrack(variant);
+    newTrack.active = true;
```

Comment:
For me this doesn't make sense here, because the one that was just changed may not be the active one due to safeMargin. @tykus160 what do you think?

Reply:
@avelad I understand your point. But according to the [docs](https://github.com/shaka-project/shaka-player/blob/e02367cc1ee1a0a4dd66426341e0be389914d7cb/externs/shaka/player.js#L294-L296), the active track is "the track being streamed (another track may be visible/audible in the buffer)". This is aligned with the current implementation ([1](https://github.com/shaka-project/shaka-player/blob/e02367cc1ee1a0a4dd66426341e0be389914d7cb/lib/media/streaming_engine.js#L453), [2](https://github.com/shaka-project/shaka-player/blob/e02367cc1ee1a0a4dd66426341e0be389914d7cb/lib/player.js#L4668-L4669), [3](https://github.com/shaka-project/shaka-player/blob/e02367cc1ee1a0a4dd66426341e0be389914d7cb/lib/player.js#L4682)).

Reply:
somewhere we should set the old one to false, and this is not doing it here...

Reply:
`shaka.util.StreamUtils.variantToTrack` set `active` to `false` by default and I've added unit tests to make sure `event.oldTrack.active` is always `false`. But I can additionally set it to `false` before firing the event. Should I add it?